services/namex-pay/src/namex_pay/worker.py

```diff
@@ -218,8 +218,25 @@ async def process_payment(pay_msg: dict, flask_app: Flask):
                             nr,
                             nr.json()
                         )
-                        # try to update NRO otherwise send a sentry msg for OPS
                         if payment.payment_action in [payment.PaymentActions.UPGRADE.value, payment.PaymentActions.REAPPLY.value]:
+                            option = 'renewal' if payment.payment_action == payment.PaymentActions.REAPPLY.value else 'upgrade'
+                            payload = create_cloud_event_msg(
+                                msg_id=str(uuid.uuid4()),
+                                msg_type='bc.registry.names.request',
+                                source=f'/requests/{nr.nrNum}',
+                                time=datetime.utcfromtimestamp(time.time()).replace(tzinfo=timezone.utc).isoformat(),
+                                identifier=nr.nrNum,
+                                json_data_body={
+                                    'request': {
+                                        'nrNum': nr.nrNum,
+                                        'option': option
+                                    }
+                                }
+                            )
```

Comment:
In https://github.com/bcgov/namex/pull/1142 I created a queue_util function to build the payload. If PR 1142 is merged before you could use it too.